clients/src/main/java/org/apache/kafka/common/requests/JoinGroupRequest.java

```diff
@@ -60,14 +62,40 @@ public ByteBuffer metadata() {
         }
     }
 
+    // v0 constructor
+    @Deprecated
```

Comment:
Do we deprecate as soon as the new version is added? Not sure this change is warranted yet. (I'm also not sure whether this is a useful indicator since this isn't public API anyway, so it should mostly only clutter our builds with warnings)